main/main.cpp

```diff
@@ -2890,7 +2890,13 @@ bool Main::iteration() {
 	frames++;
 	Engine::get_singleton()->_process_frames++;
 
-	if (frame > 1000000) {
+	// Smoothly update FPS over time, while ensuring the speed at which FPS stabilizes is not dependent on FPS.
+	// The FPS smooth factor is tuned to always *mostly* converge within a second.
+	constexpr int FPS_SMOOTH_FACTOR = 5;
+	const double ratio = FPS_SMOOTH_FACTOR * USEC_TO_SEC(frame_time);
```

Comment:
```suggestion
	const double ratio = FPS_SMOOTH_FACTOR * USEC_TO_SEC(frame_time);
```
ticks_elapsed should work much better than frame_time :) (Obviously same on line below)